cf-agent/verify_users_pam.c

```diff
@@ -1690,6 +1684,19 @@ static bool DoModifyUser (const char *puser, const User *u, const struct passwd
         }
     }
 
+    if (CFUSR_CHECKBIT (changemap, i_groups) != 0)
+    {
+        StringAppend(cmd, " -G \"", sizeof(cmd));
```

Comment:
Here we set the secondary groups. It could be an empty set and result in `-G ""` and so far in testing on debian, freebsd, suse and hp-ux I haven't seen any problems with this as mentioned in previous version of the code.